discretisedfield/field.py

```diff
@@ -109,15 +109,19 @@ class Field:
 
     """
 
-    def __init__(self, mesh, dim, value=0, norm=None, components=None):
+    def __init__(self, mesh, dim, value=0, norm=None, components=None,
+                 dtype=np.float64):
         self.mesh = mesh
         self.dim = dim
-        self.value = value
-        self.norm = norm
+        self.dtype = dtype
 
+        # order in here is important to avoid infinite recursion
```

Comment:
I am not sure I understand what is the recursion problem here. We could probably rewrite component setter.

Reply:
The recursion problem disappeared upon rewriting as_array